sound/soc/codecs/rt5682-sdw.c

```diff
@@ -717,7 +717,7 @@ static int rt5682_sdw_remove(struct sdw_slave *slave)
 }
 
 static const struct sdw_device_id rt5682_id[] = {
-	SDW_SLAVE_ENTRY_EXT(0x025d, 0x5682, 0x1, 0, 0),
+	SDW_SLAVE_ENTRY(0x025d, 0x5682, 0),
```

Comment:
I don't understand this change @bardliao 
The extended definition is backwards compatible with the previous one, so it doesn't change anything if there is no version or class information?
what are you trying to fix?

Reply:
If I read #2161 the correct fix should be

SDW_SLAVE_ENTRY_EXT(0x025d, 0x5682, 0x2, 0, 0)

No need to remove the entire macro, just add the correct version number

Reply:
> If I read #2161 the correct fix should be
> 
> SDW_SLAVE_ENTRY_EXT(0x025d, 0x5682, 0x2, 0, 0)
> 
> No need to remove the entire macro, just add the correct version number

Yes, I thought about it. But I am not sure if sdw version 1.0 and 1.1 of rt5682 are coexisting or not. Realtek told me at least for now there is no plan to support sdw 1.2 so there will not be a driver for rt5682 with sdw version 1.2. That's why I revert the change of using SDW_SLAVE_ENTRY_EXT.

Reply:
i don't think there is a 1.0 version?